### PR TITLE
chore(e2e):  optimize .ibm Dockerfile for Improved Build Performance and Dependency Updates

### PR DESCRIPTION
**Pull request by gustavolira**
https://issues.redhat.com/browse/RHIDP-4956

Optimized the Dockerfile by combining multiple RUN commands into a single layer to reduce the size and improve build efficiency. Set essential environment variables together, consolidated installations, and added comments for clarity. This refactor enhances maintainability and performance of the Docker build process.

### CODE REVIEW

**Comment by nickboldt** [comment]
Would be cool if you could also switch this to be based on UBI9 so we're not using Ubuntu to test RH product :D

**Comment by zdrapela** [comment]
> Would be cool if you could also switch this to be based on UBI9 so we're not using Ubuntu to test RH product :D 

Unfortunately, Playwright only supports Ubuntu and Debian from Linux distros 😢

File: .ibm/images/Dockerfile [hunk 1]
```diff
@@ -1,103 +1,62 @@
...
+RUN curl -fsSL "https://github.com/mikefarah/yq/releases/download/v4.43.1/yq_linux_amd64.tar.gz" | tar -xz && \
+    mv yq_linux_amd64 /usr/local/bin/yq && \
+    curl -sL https://aka.ms/InstallAzureCLIDeb | bash && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
+# Install additional tools and IBM Cloud CLI plugins
 RUN go install github.com/kadel/pr-commenter@latest && \
     ibmcloud plugin install -f cloud-object-storage && \
-    ibmcloud plugin install -f kubernetes-service
+    ibmcloud plugin install -f kubernetes-service && \
```
Comment (zdrapela) [inline .ibm/images/Dockerfile:58]: Why not join the `Install IBM Cloud CLI` with `IBM Cloud CLI plugins`?

File: .ibm/images/Dockerfile [hunk 2]
```diff
@@ -1,103 +1,62 @@
...
+    ibmcloud --version
 
-# Set environment variables to make Go work correctly
-ENV GOPATH /go
-ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
+# Install yq and Azure CLI
+RUN curl -fsSL "https://github.com/mikefarah/yq/releases/download/v4.43.1/yq_linux_amd64.tar.gz" | tar -xz && \
+    mv yq_linux_amd64 /usr/local/bin/yq && \
+    curl -sL https://aka.ms/InstallAzureCLIDeb | bash && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```
Comment (zdrapela) [inline .ibm/images/Dockerfile:53]: Why not keep Azure CLI alone and join `yq` with `# Install Go, Helm, OpenShift CLI, and ocm-cli`?

**Review by zdrapela — COMMENTED** [review]
Hi @gustavolira, 
have you measured how much that speeds up the build process?
Some of the optimization is definitely needed and helps with speeding up, but I believe that having too many `RUN` commands squashed together reduces the maintainability and readability of the `Dockerfile`.
The `# Install essential dependencies and Node.js tools` and `# Install Go, Helm, OpenShift CLI, and ocm-cli` seem to be quite packed up.

File: .ibm/images/Dockerfile [hunk 3]
```diff
@@ -1,103 +1,62 @@
...
-ENV GO_SHA256 464b6b66591f6cf055bc5df90a9750bf5fbc9d038722bb84a9d56a2bea974be6
-
-# Install Go and other tools used by the pipeline
-RUN apt-get update && \
-    apt-get install -y curl && \
-    curl -LO "https://golang.org/dl/go${GO_VERSION}.linux-amd64.tar.gz" && \
-    echo "${GO_SHA256} go${GO_VERSION}.linux-amd64.tar.gz" | sha256sum -c - && \
+# Install Go, Helm, OpenShift CLI, ocm-cli, and yq
+RUN curl -LO "https://golang.org/dl/go${GO_VERSION}.linux-amd64.tar.gz" && \
+    echo "464b6b66591f6cf055bc5df90a9750bf5fbc9d038722bb84a9d56a2bea974be6 go${GO_VERSION}.linux-amd64.tar.gz" | sha256sum -c - && \
```
Comment (zdrapela) [inline .ibm/images/Dockerfile:37]: I think that the `GO_SHA256` should be kept as a variable close to `GO_VERSION`. You would be updating them both together. The comment `# Set Go version and the expected SHA256 hash for verification` seems to be useful and I would keep it around.

**Comment by zdrapela** [comment]
I added one more comment regarding Go. Apart from that I'm good with the changes once the PR is green

File: .ibm/images/Dockerfile [hunk 4]
```diff
@@ -1,103 +1,62 @@
...
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
 # Install Azure CLI
-RUN curl -sL https://aka.ms/InstallAzureCLIDeb | bash
-
-# Set environment variables to make Go work correctly
-ENV GOPATH /go
-ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
-
-RUN go install github.com/kadel/pr-commenter@latest && \
```
Comment (zdrapela) [inline .ibm/images/Dockerfile:99]: Is this safe to remove, or is it used somewhere? @kadel

Comment (kadel) [reply .ibm/images/Dockerfile:99]: If I recall correctly, this was used to post PR comment with logs in CI jobs running on IBM cloud.
If nothing is running there anymore, this should be safe to delete.

Comment (josephca) [reply .ibm/images/Dockerfile:99]: IIRC, the pr commenter was used a long time ago and no longer the case. So, it can be removed safely.

Comment (gustavolira) [reply .ibm/images/Dockerfile:102]: removed `GO` and `pr-commenter`

**Review by josephca — APPROVED** [review]
/lgtm

**Pull request merged**